src/huggingface_hub/utils/_typing.py

```diff
@@ -14,7 +14,9 @@
 # limitations under the License.
 """Handle typing imports based on system compatibility."""
 
-from typing import Any, Callable, Literal, TypeVar
+from itertools import filterfalse
+from types import UnionType
```

Comment:
Looks like `UnionType` is not available on Python 3.8 :confused:

Reply:
😢 made the updates!